File: crawl_module/domain/company.py
from datetime import datetime

from peewee import *

from crawl_module.domain.base_model import BaseModel


class TbCompany(BaseModel):
    company_id = BigAutoField()
    company_name = CharField(max_length=255)
    industry_type = CharField(max_length=255, null=True)
    company_type = CharField(max_length=255, null=True)
    employee_count = IntegerField(null=True)
    address = CharField(max_length=255, null=True)
    homepage = CharField(max_length=255, null=True)
    phone_number = CharField(max_length=255, null=True)
    thumbnail_url = CharField(max_length=255, null=True)
    created_at = DateTimeField(default=datetime.now)
    updated_at = DateTimeField(default=datetime.now)

    class Meta:
        db_table = 'TB_COMPANY'

File: src/main/java/com/why_not_cote/entity/company/Company.java
package com.why_not_cote.entity.company;

import com.why_not_cote.entity.CommonBaseDateTime;
import com.why_not_cote.entity.post.HirePost;
import java.util.List;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Table(name = "TB_COMPANY")
@NoArgsConstructor
public class Company extends CommonBaseDateTime {
    @Id
    @Column(name = "company_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany(mappedBy = "company")
    private List<HirePost> postList;

    @Column(name = "origin_company_id")
    private String originCompanyId;

    @Column(name = "company_name")
    private String companyName;

    @Column(name = "industry_type")
    private String industryType;

    @Column(name = "company_type")
    private String companyType;

    @Column(name = "employee_count")
    private Long employeeCount;

    @Column(name = "address")
    private String address;

    @Column(name = "homepage")
    private String homepage;

    @Column(name = "phone_number")
    private String phoneNumber;

    @Column(name = "thumbnail_url")
    private String thumbnailUrl;

    @Builder
    public Company(String originCompanyId, String companyName,
        String industryType, String companyType, Long employeeCount, String address,
        String homepage,
        String phoneNumber, String thumbnailUrl) {
        this.originCompanyId = originCompanyId;
        this.companyName = companyName;
        this.industryType = industryType;
        this.companyType = companyType;
        this.employeeCount = employeeCount;
        this.address = address;
        this.homepage = homepage;
        this.phoneNumber = phoneNumber;
        this.thumbnailUrl = thumbnailUrl;
    }
}
